### PR TITLE
Allow overriding timestamp of event

### LINKED ISSUE #42: Allow overriding event timestamps
Currently, event timestamps are set at time of recording, and can not be changed.

I'm writing a small tool that'll generate jupyter_telemetry style outputs by parsing JupyterHub logs, and not being able to override the timestamp makes this impossible currently.

### PR DESCRIPTION
**Pull request by yuvipanda**
Very useful when generating eventlog streams from pre-recorded
data.

Fixes #42

### CODE REVIEW

File: jupyter_telemetry/eventlog.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def register_schema(self, schema):
 
         self.schemas[(schema['$id'], schema['version'])] = schema
 
-    def record_event(self, schema_name, version, event):
```
Comment (Zsailer) [inline jupyter_telemetry/eventlog.py]: I think this method's signature is getting big enough now that we might want to add documentation for each argument to the docstring... It doesn't have to be in this PR. I was thinking about doing a documentation audit sometime soon.

**Comment by Zsailer** [comment]
Thanks, @yuvipanda! This looks great! 

Quick follow up question—is it dangerous effect/override timestamps of events? I think, as long as we're not exposing this functionality to operators and users (i.e. through some configurable option etc), it should be safe, right? We don't want people changing timestamps to obfuscate their activity in some way.

**Comment by yuvipanda** [comment]
https://github.com/yuvipanda/jupyterhub-telemetry-extractor is the code I wrote! Pushes out telemetry logs from parsing JupyterHub logs.

File: jupyter_telemetry/eventlog.py (hunk 1 above)
Comment (yuvipanda) [reply jupyter_telemetry/eventlog.py]: +1 for documentation audit soon!

**Comment by yuvipanda** [comment]
@Zsailer I don't think this PR changes our security posture. For examples, admins can put in loggers with custom handlers that can change timestamps (or any other data) as they wish. If users are running code in the same python process (as we do with notebooks most of the time), they can also monkeypatch easily to do the same thing. So we should continue treating all telemetry from user created processes as untrusted, regardless of what happens with this PR...

**Comment by Zsailer** [comment]
> we should continue treating all telemetry from user created processes as untrusted

:+1: Sounds good. This looks good to me!

**Pull request merged**